lib/node_modules/@stdlib/ndarray/some-by/README.md

```diff
@@ -0,0 +1,389 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2025 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# someBy
+
+> Test whether at least `n` elements in an [`ndarray`][@stdlib/ndarray/ctor] pass a test implemented by a predicate function along one or more dimensions.
+
+<section class="intro">
+
+</section>
+
+<!-- /.intro -->
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var someBy = require( '@stdlib/ndarray/some-by' );
+```
+
+#### someBy( x, n\[, options], predicate\[, thisArg] )
+
+Tests whether at least `n` elements in an [`ndarray`][@stdlib/ndarray/ctor] pass a test implemented by a predicate function along one or more dimensions.
+
+<!-- eslint-disable max-len -->
+
+```javascript
+var Float64Array = require( '@stdlib/array/float64' );
+var ndarray = require( '@stdlib/ndarray/ctor' );
+
+function predicate( value ) {
+    return value > 0.0;
+}
+
+// Create a data buffer:
+var xbuf = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0 ] );
+
+// Define the shape of the input array:
+var sh = [ 3, 1, 2 ];
+
+// Define the array strides:
+var sx = [ 4, 4, 1 ];
+
+// Define the index offset:
+var ox = 1;
+
+// Create an input ndarray:
+var x = new ndarray( 'float64', xbuf, sh, sx, ox, 'row-major' );
+
+// Perform reduction:
+var out = someBy( x, 2, predicate );
+// returns <ndarray>
+
+console.log( out.get() );
+// => true
+```
+
+The function accepts the following arguments:
+
+-   **x**: input [`ndarray`][@stdlib/ndarray/ctor].
+-   **n**: number of elements which must pass the test implemented by a predicate function. May be either a scalar or an [`ndarray`][@stdlib/ndarray/ctor].
```

Comment:
As in `blas/ext/cusum`, you need to explicitly state that `n` must be broadcast-compatible with `x`.

```suggestion
-   **n**: number of elements which must pass the test implemented by a predicate function. May be either a scalar or an [`ndarray`][@stdlib/ndarray/ctor]. Must be [broadcast-compatible][@stdlib/ndarray/base/broadcast-shapes] with the non-reduced dimensions of input [`ndarray`][@stdlib/ndarray/ctor].
```